DB設計

yukri1030 opened the pull request:
＃WHAT
DBの設計をしました。
memberを中間テーブルとして設計しました

# WHY
何か足りない部分があれば、コメントお願いいします

yukri1030 committed:
[WPI]DB設計

＃WHAT
DBの設計をしました。
memberを中間テーブルとして設計しました

# WHY
何か足りない部分があれば、コメントお願いいします

miyagit commented on README.md:
```diff
@@ -1,24 +1,58 @@
+- belongs_to: user 
+
+
+gem 'haml-rails'
```
こちらなにでしょう？

yukri1030 replied:
返信ありがとうございます！上記のgemは削除しました。

##  何
メンバーテーブル（中間テーブル）に
has_many：messages; を追加

メッセージテーブルに
belongs_to：member; を追加しました。

あとは、スペルミスを見直し、gemを消しました

## なぜ
足りない部分があれば教えていたいだいたいです。

miyagit reviewed (commented):
中間テーブルのアソシエーションが足りないものがあります。アソシエーションを見直しましょう。

yukri1030 committed:
[WPI]DB設計見直し

# WHAT
memberテーブル（中間テーブル）に
has_many: mwssages;  を追加

messageーテーブルに
belongs_to: member;  を追加しました。

あとは、スペルミスを見直し、gemを消しました

# WHY
足りない部分があれば教えていただきたいです。

shingoteshima commented on README.md:
```diff
@@ -1,24 +1,59 @@
+## usersテーブル
+|Column|Type|Opitions|
+|------|----|---------|
+|id|integer|null:false, foreign_key: true|
```
idは自動生成されますので書かなくて大丈夫です。

yukri1030 replied:
ありがとうございます！
group_id
member_id
も消してしまって大丈夫ですか？

yukri1030 replied:
間違えました。
groupテーブルのid
memberテーブルのid
も消してしまって良いですか

shingoteshima replied:
消しましょう。

yukri1030 replied:
ありがとうございます！消しました！

shingoteshima commented on README.md:
```diff
@@ -1,24 +1,59 @@
+|------|----|---------|
+|id|integer|null:false, foreign_key: true|
+|------|----|---------|
+|name|string|nill:false, foreign_key: true, uniqe:true|
```
nillではなくnullですね。

yukri1030 replied:
修正します！

shingoteshima commented on README.md:
```diff
@@ -1,24 +1,59 @@
+|id|integer|null:false, foreign_key: true|
+|------|----|---------|
+|name|string|nill:false, foreign_key: true, uniqe:true|
+|e-mail|varchar(30)|null:false|
```
こちらはdeviseを使えば自動生成されますので消してしまって大丈夫です。

shingoteshima commented on README.md:
```diff
@@ -1,24 +1,59 @@
+|------|----|---------|
+|name|string|nill:false, foreign_key: true, uniqe:true|
+|e-mail|varchar(30)|null:false|
+|message_id|null:false, foreign_key: true|
```
has_many :messagesのためmessage_idは不要です。

yukri1030 replied:
修正します！しかし、正直、message_idが不要な理由がよくわkリマセン。

shingoteshima replied:
user has_many :messagesだからです。

yukri1030 replied:
has_manyのアソシエーションがあると、idで紐付けなくても、参照されるのでしょうか

shingoteshima replied:
ここは一対多の関係なので、一度復習してください。
pictweetで復習されれば理解できると思います。

shingoteshima commented on README.md:
```diff
@@ -1,24 +1,59 @@
+|name|string|nill:false, foreign_key: true, uniqe:true|
+|e-mail|varchar(30)|null:false|
+|message_id|null:false, foreign_key: true|
+|member_id|integer|null:false, foreign_key: true|
```
has_many :membersなのでmember_idは不要です。

yukri1030 replied:
member_id消しました！

shingoteshima commented on README.md:
```diff
@@ -1,24 +1,59 @@
+### Asociation
+- has_many :groups , through :member
+- has_many :messages
+
```
has_many :membersを追加しましょう。

shingoteshima replied:
has_many :members追加してください。

shingoteshima commented on README.md:
```diff
@@ -1,24 +1,59 @@
+
+|Column|Type|Opitions|
+|------|----|---------|
+|id|integer|null:false, foreign_key: true｜
```
idは自動生成されますので書かなくて大丈夫です。

shingoteshima commented on README.md:
```diff
@@ -1,24 +1,59 @@
+|member_id|integer|null:false, foreign_key: true|
+
+### Asociation
+- has_many: members through: users
```
groupsテーブルのアソシエーションは
has_many :members
has_many :users, through: :members
has_many :messages
となります。

shingoteshima commented on README.md:
```diff
@@ -1,24 +1,59 @@
+|------|----|---------|
+|id|integer|null:false, foreign_key: true｜
+|name|string|null: false,uniqe:true|
+|member_id|integer|null:false, foreign_key: true|
```
has_many :membersなのでmember_idは不要です。

shingoteshima replied:
member_idは不要です。

shingoteshima replied:
member_idは必要ですか？

shingoteshima commented on README.md:
```diff
@@ -1,24 +1,59 @@
+
+|Column|Type|Opitions|
+|------|----|---------|
+|id|integer|null:false, foreign_key: true|
```
idは自動生成されますので書かなくて大丈夫です。

yukri1030 replied:
消しました！ありがとうございます！

shingoteshima commented on README.md:
```diff
@@ -1,24 +1,59 @@
-* System dependencies
+- belongs_to: group 
+- belongs_to: user
+- has_many: messages
```
has_many: messagesは不要です。

shingoteshima replied:
has_many: messages消しましょう。

shingoteshima commented on README.md:
```diff
@@ -1,24 +1,59 @@
+|Column|Type|Opitions|
+|------|----|---------|
+|id|integer|null:false, foreign_key: true|
+|user_id|string|null: false,foreign_key :true|
```
stringではなくreferencesにしましょう。
参考: https://qiita.com/ryouzi/items/2682e7e8a86fd2b1ae47

shingoteshima commented on README.md:
```diff
@@ -1,24 +1,59 @@
+|------|----|---------|
+|id|integer|null:false, foreign_key: true|
+|user_id|string|null: false,foreign_key :true|
+|group_id|integer|null:false, foreign_key: true|
```
integerではなくreferencesにしましょう。

yukri1030 replied:
reference型に直しました

shingoteshima commented on README.md:
```diff
@@ -1,24 +1,59 @@
-* Database initialization
+|Column|Type|Opitions|
+|------|----|---------|
+|id|integer|null:false, foreign_key: true｜
```
idは自動生成されますので書かなくて大丈夫です。

yukri1030 replied:
messageテーブル
groupテーブル
memberテーブルの
それぞれのidを削除しても大丈夫ですか

shingoteshima replied:
削除して大丈夫です。

shingoteshima commented on README.md:
```diff
@@ -1,24 +1,59 @@
+|id|integer|null:false, foreign_key: true｜
+|message|text|
+|img|string|
 
```
belongs_to: user なのでuser_idをもたせましょう。

shingoteshima commented on README.md:
```diff
@@ -1,24 +1,59 @@
 
-* Services (job queues, cache servers, search engines, etc.)
+- belongs_to: user 
+- belongs_to: member
```
belongs_to: memberではなく、belongs_to :groupではないでしょうか？

yukri1030 replied:
mssageテーブルなのですが、belongs_to: group ;なのでしょうか

shingoteshima replied:
messageはgroupに属しているため、belongs_to :groupとなります。

yukri1030 replied:
修正します！正直、まだ理解できなくて。groupがmessageを投稿することはないと思うのですが・・・

shingoteshima replied:
一対多の関係を復習しましょう。

yukri1030 committed:
DB設計直し

## WHAT
usersテーブルのカラムは、全てなくなりました。

user_idは、referance型にしました。

## Why
お願いします

yukri1030 reviewed (commented):
50行目のidを削除するのですか。

yukri1030 committed:
DB 見直し

## WHAT
それぞれのテーブルの, idを削除しました！
messageテーブルでは、
belongs_to: user;にしました

 ## WHY
messageテーブルのimgカラムは、stringで大丈夫ですか。

yukri1030 commented:
アソシエーションなどは復習しようと思います！
他、コードは大丈夫ですか

yukri1030 committed:
DB設計

## WHAT
アソシエーションを見直しました

## WHY
他は、修正する点はありますか

shingoteshima commented on README.md:
```diff
@@ -1,24 +1,51 @@
+## usersテーブル
+|Column|Type|Opitions|
+|------|----|---------|
+
```
nameを追加しましょう。

shingoteshima commented on README.md:
```diff
@@ -1,24 +1,51 @@
+|member_id|integer|null:false, foreign_key: true|
+
+### Asociation
+- has_many: users through: members
```
アソシエーションがたりません。

TOMARUMARU replied:
別途membersとも組む必要があります

shingoteshima commented on README.md:
```diff
@@ -1,24 +1,51 @@
+|Column|Type|Opitions|
+|------|----|---------|
+|user_id|reference|null: false,foreign_key :true|
+|group_id|integer|null:false, foreign_key: true|
```
こちらもreferencesにしましょう。

shingoteshima commented on README.md:
```diff
@@ -1,24 +1,51 @@
+|------|----|---------|
+|message|text|
+|img|string|
 
```
カラムが足りません。
user_id, group_id

yukri1030 committed:
DB見直し

## WHAT
userテーブルに、nameカラムを追加しました
アソシエーションは、has_many: members;を追加しました

groupテーブル
- has_many: members through: users

を追加しました

memberテーブル
|group_id|reference||null:false, foreign_key: true|

reference型にしました

messageテーブル
|user_id|reference|null: false,foreign_key :true|
|group_id|reference||null:false, foreign_key: true|

設定しました

shingoteshima commented on README.md:
```diff
@@ -1,24 +1,55 @@
+
+
+### Asociation
+- has_many :groups , through :member
```
through :membersと複数形です。

shingoteshima commented on README.md:
```diff
@@ -1,24 +1,55 @@
+
+### Asociation
+- has_many: users through: members
+- has_many: members through: users
```
through: usersは不要です。

shingoteshima commented on README.md:
```diff
@@ -1,24 +1,55 @@
+### Asociation
+- has_many: users through: members
+- has_many: members through: users
+
```
messagesテーブルとのアソシエーションがありません。

shingoteshima replied:
messagesテーブルとのアソシエーションがありません。こちら修正しましょう。

yukri1030 committed:
DB設計見直し

## WHAT
membersと複数形にしました

groupテーブルのアソシエーションを追記しました

## WHY
アソシエーションは大丈夫ですか

また、groupテーブルに
中間テーブルのmember_idが入っているのですが、大丈夫ですか

shingoteshima commented on README.md:
```diff
@@ -1,24 +1,55 @@
+
+### Asociation
+- has_many: users through: members
+- has_many: messages through: users
```
through: users不要です。

TOMARUMARU replied:
こちら修正してください

yukri1030 committed:
DB直し

## WHAT
groupテーブルの、user_idを削除しました

yukri1030 committed:
DB見直し

## WHAT
group,  user  memberの３つのテーブルに、
messages_id|reference|null: false,foreign_key :true|を追加しました

## WHY
上記３つのテーブル全てを紐付ける中間テーブルは、memberにした方が、分かりやすいでしょうか。
（つまり、memberテーブルに、message_id）を付け加えるべきですか

正直、中間テーブルの理解ができていません。中間テーブルを経て紐づいているテーブル同士は、その対のテーブルのidをカラムに入れなくても良いのでようか

http://matsuda-juri.hatenablog.com/entry/2017/05/16/012457

TOMARUMARU commented on README.md:
```diff
@@ -1,24 +1,58 @@
+|Column|Type|Opitions|
+|------|----|---------|
+|name|string|null: false,uniqe:true|
+|messages_id|reference|null: false,foreign_key :true|
```
message_idは不要です

TOMARUMARU commented on README.md:
```diff
@@ -1,24 +1,58 @@
+|Column|Type|Opitions|
+|------|----|---------|
+|name|string|null: false,uniqe:true|
+|messages_id|reference|null: false,foreign_key :true|
```
message_idは不要です

TOMARUMARU commented on README.md:
```diff
@@ -1,24 +1,58 @@
+|------|----|---------|
+|user_id|reference|null: false,foreign_key :true|
+|group_id|reference||null:false, foreign_key: true|
+|messages_id|reference|null: false,foreign_key :true|
```
message_idは不要です

TOMARUMARU reviewed (commented):
Files changedを確認して指摘されている箇所を全て確認してください

yukri1030 committed:
DBやり直し

## WHAT
データが消えてしまったため、アドバイスを参考にまた作り直しました。

## WHY
ソシエーションは大丈夫でしょうか

TOMARUMARU commented:
すみませんどういうことでしょうか？
差分がなくなってしまっていますが…

yukri1030 commented:
すみません・・・一回、全てのコードを削除してしまったのです。空のREAD.meのファイル上で、新しくコードを書きました。そのため、差分がなくなってしまったものと思われます。
